src/main/java/internet/shop/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package internet.shop.service.impl;
+
+import internet.shop.dao.ShoppingCartDao;
+import internet.shop.lib.Inject;
+import internet.shop.lib.Service;
+import internet.shop.model.Product;
+import internet.shop.model.ShoppingCart;
+import internet.shop.service.ShoppingCartService;
+import java.util.List;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+
+    public List<Product> getProducts(ShoppingCart shoppingCart) {
+        return shoppingCartDao.getProducts(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        return shoppingCartDao.create(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
+        shoppingCart.getProducts().add(product);
```

Comment:
we need to call update method after, for shoppingCart in storage to receive changes, made here 
same for clear method